migration/migrate.go

```diff
@@ -0,0 +1,142 @@
+package migration
+
+import (
+	"errors"
+	"os"
+	"strings"
+	"text/template"
+	"time"
+
+	"gofr.dev/pkg/gofr"
+)
+
+const (
+	mig     = "migrations"
+	allFile = "all.go"
+)
+
+var (
+	errNameEmpty = errors.New(`please provide the name of the migration using "-name" option`)
+)
+
+//nolint:gochecknoglobals // keeping them local so that they are computed at the compile time.
+var (
+	allTemplate = template.Must(template.New("allContent").Parse(
+		`// This is auto-generated file using 'gofr migrate' tool. DO NOT EDIT.
+package migrations
+
+import (
+	"gofr.dev/pkg/gofr/migration"
+)
+
+func All() map[int64]migration.Migrate {
+	return map[int64]migration.Migrate {
+{{range $key, $value := .}}	
+		{{ $key }}: {{ $value }}(),{{end}}
+	}
+}
+`))
+
+	migrationTemplate = template.Must(template.New("migrationContent").
+				Parse(
+			`package migrations
+
+import (
+	"gofr.dev/pkg/gofr/migration"
+)
+
+func {{ . }}() migration.Migrate {
+	return migration.Migrate{
+		UP: func(d migration.Datasource) error {
+			// write your migrations here
+
+			return nil
+		},
+	}
+}
+`))
+)
+
+func Migrate(ctx *gofr.Context) (interface{}, error) {
+	migName := ctx.Param("name")
+	if migName == "" {
+		return nil, errNameEmpty
+	}
+
+	if err := createMigrationFile(ctx, migName); err != nil {
+		return nil, err
```

Comment:
It will be better if we provide a more detailed error like :
`fmt.Errorf("failed to create migration file: %w", err)`